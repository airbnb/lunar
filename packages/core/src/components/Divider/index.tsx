import React from 'react';
import withStyles, { WithStylesProps } from '../../composers/withStyles';
import Spacing, { SpacingRange } from '../Spacing';

export type Props = {
  /** Spacing on the bottom. */
  bottom?: SpacingRange;
  /** Spacing on the top. */
  top?: SpacingRange;
};

/** A horizontal divider. */
export class Divider extends React.Component<Props & WithStylesProps> {
  static defaultProps: Props = {
    bottom: 2,
    top: 2,
  };

  render() {
    const { cx, styles, bottom, top } = this.props;

    return (
      <Spacing bottom={bottom} top={top}>
        <div className={cx(styles.divider)} />
      </Spacing>
    );
  }
}

export default withStyles(({ ui }) => ({
  divider: {
    borderBottom: ui.border,
  },
}))(Divider);
