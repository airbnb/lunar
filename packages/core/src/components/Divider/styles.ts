import { StyleSheet } from '../../hooks/useStyles';

const styleSheet: StyleSheet = ({ ui, unit }) => ({
  divider: {
    borderBottom: ui.border,
  },

  divider_short: {
    width: 4 * unit,
  },
});

export default styleSheet;

export { styleSheet };
