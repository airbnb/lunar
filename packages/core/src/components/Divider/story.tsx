import React from 'react';
import Divider from '.';

export default {
  title: 'Core/Divider',
  parameters: {
    inspectComponents: [Divider],
  },
};

export function standardDivider() {
  return <Divider />;
}

standardDivider.story = {
  name: 'Standard divider.',
};

export function shortDivider() {
  return <Divider short />;
}

shortDivider.story = {
  name: 'Short divider.',
};
