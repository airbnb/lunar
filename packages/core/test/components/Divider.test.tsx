import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import Divider from '../../src/components/Divider';

describe('<Divider />', () => {
  it('renders a divider', () => {
    const wrapper = shallowWithStyles(<Divider />);

    expect(wrapper).toMatchSnapshot();
  });

  it('can pass top and bottom to spacing', () => {
    const wrapper = shallowWithStyles(<Divider bottom={4} top={2} />);

    expect(wrapper.prop('bottom')).toBe(4);
    expect(wrapper.prop('top')).toBe(2);
  });
});
