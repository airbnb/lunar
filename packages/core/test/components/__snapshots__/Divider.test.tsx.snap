// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> renders a divider 1`] = `
<Spacing
  bottom={2}
  top={2}
>
  <div
    className="divider_bi7c4v"
  />
</Spacing>
`;
